File: frontend/lib/models/appointment.dart
class Appointment {
  final String id;
  final String date;
  final String time;
  final String status;
  final String? trainerId;
  final String? trainerName;
  final String? userId;
  final String? userName;
  final String? userEmail;
  final String? notes;
  final String scolor;
  final double price;

  Appointment({
    required this.id,
    required this.date,
    required this.time,
    required this.status,
    this.trainerId,
    this.trainerName,
    this.userId,
    this.userName,
    this.userEmail,
    this.notes,
    this.scolor = '#FF7F27',
    this.price = 50.0,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['_id'],
      date: json['date'],
      time: json['time'],
      status: json['status'],
      trainerId: json['trainer'] is Map ? json['trainer']['_id'] : json['trainer'],
      trainerName: json['trainer'] is Map ? json['trainer']['name'] : 'Unknown Trainer',
      userId: json['user'] is Map ? json['user']['_id'] : json['user'],
      userName: json['user'] is Map ? json['user']['name'] : 'Unknown User',
      userEmail: json['user'] is Map ? json['user']['email'] : null,
      notes: json['notes'],
      scolor: json['scolor'] ?? '#FF7F27',
      price: (json['price'] ?? 50.0).toDouble(),
    );
  }
}

File: backend/models/Appointment.js
const mongoose = require('mongoose');

const appointmentSchema = mongoose.Schema({
    user: {
        type: mongoose.Schema.Types.ObjectId,
        required: true,
        ref: 'User'
    },
    trainer: {
        type: mongoose.Schema.Types.ObjectId,
        required: [true, 'Fadlan dooro tababare (Please select a trainer)'],
        ref: 'User'
    },
    date: {
        type: String, // Waxaan isticmaali karnaa Date object, laakiin String ayaa ka fudud hadda.
        required: [true, 'Fadlan dooro taariikh (Please select a date)']
    },
    time: {
        type: String,
        required: [true, 'Fadlan dooro waqti (Please select a time)']
    },
    status: {
        type: String,
        enum: ['pending', 'confirmed', 'cancelled', 'completed'],
        default: 'pending'
    },
    notes: {
        type: String,
        required: false
    },
    price: {
        type: Number,
        default: 50 // Default price per session
    },
    scolor: {
        type: String,
        default: '#FF7F27' // Default to AppColors.primary (Orange)
    }
}, {
    timestamps: true
});

module.exports = mongoose.model('Appointment', appointmentSchema);

//this appinment model

